test/js/node/watch/fs.watch.test.ts

```diff
@@ -97,6 +102,67 @@ describe("fs.watch", () => {
     });
   });
 
+  test("should watch new-created files", done => {
+    const root = path.join(testDir, "test-newfile-directory");
+    try {
+      fs.mkdirSync(root);
+      fs.writeFileSync(path.join(root, "test-started.txt"), "");
+    } catch {}
+    let count = 0;
+    let ready = false;
+    let err: Error | undefined = undefined;
+    const watcher = fs.watch(root, { signal: AbortSignal.timeout(3000) });
+    watcher.on("change", (event, filename) => {
+      // There might be multiple events when the producer
+      // is not aware of the changes in "ready" status.
+      if (ready && filename === "test-started.txt") {
+        return;
+      }
+      count++;
+      try {
+        if (count == 1) {
+          ready = true;
+        } else if (count == 2) {
+          expect(event).toBe("rename");
+          expect(filename).toBe("new-file.txt");
+        } else if (count == 3) {
+          expect(event).toBe("change");
```

Comment:
It seems that on darwin platform, `onFSEventUpdate` is being executed instead of `onFileUpdate`. Unfortunately, I don't have a Mac laptop at the moment to test this 🤦. Feel sorry about this.

https://github.com/oven-sh/bun/blob/2dc90f39081131f8994dbf687f131cfa2105ad25/src/bun.js/node/node_fs_watcher.zig#L271-L288

Reply:
No worries if needed I fix macOS after this